### PR TITLE
feat(evmutil): add allow list for evm-convertible sdk denoms

### PR DESCRIPTION
**Pull request by pirtleshell**
## Description
adds `allowed_native_denoms` to evmutil params. the param has validation tests and is included in the module's genesis state but is currently unused. the parameter will be used to allow certain sdk denoms to be converted to erc20s in the evm. in addition to the sdk denom, it includes metadata for the erc20 token that will be deployed.

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

File: x/evmutil/types/conversion_pair.go [hunk 1]
```diff
@@ -87,3 +92,82 @@ func validateConversionPairs(i interface{}) error {
...
+// NewAllowedNativeCoinERC20Token returns an AllowedNativeCoinERC20Token
+func NewAllowedNativeCoinERC20Token(
+	sdkDenom, name, symbol string,
+	decimal uint32,
+) AllowedNativeCoinERC20Token {
+	return AllowedNativeCoinERC20Token{
+		SdkDenom: sdkDenom,
+		Name:     name,
+		Symbol:   symbol,
+		Decimal:  decimal,
```
Comment (nddeluca) [inline x/evmutil/types/conversion_pair.go:109]: Decimal must fit into a uint8 (https://eips.ethereum.org/EIPS/eip-20) so we should validate it's `< 256`

**Review by nddeluca — CHANGES_REQUESTED** [review]
Looks good.

Just the small change of `decimal -> decimals` might be good.  It can be confusing when interfaces don't match exactly, but just a nit.

Do think we should add the check for a value less than 256.  Though, I actually don't know what the behavior would be.  Will the contract deployment fail, or will solidity do an implicit conversion and overflow?

File: x/evmutil/types/conversion_pair.go (hunk 1 above)
Comment (pirtleshell) [reply x/evmutil/types/conversion_pair.go:109]: ah yup. i'll just make it a uint8 like in the erc20 standard.
will also pluralize 👍

**Review by nddeluca — CHANGES_REQUESTED** [review]
Decimal changes look great.  Like the usage of `math.MaxUint8` 👍 

One more uncertainty here:

We are modifying parameters, which is a consensus breaking change.

What's the behavior of upgrading with default (empty) upgrade handler, is the default set appropriately? Does GetParams work after upgrade?
Do historical GetParams work and handle the missing value gracefully (without panic)?

Can we write unit tests to cover those cases above?

Once we have those tests, should we consider bumping the consensus version in this PR and add a migration?

**Comment by pirtleshell** [comment]
alrighty @nddeluca, without any migrations the behavior is such that the chain will upgrade, but the param indeed never gets set in the store.

upgrading from v0.23.0 to this branch with an empty upgrade handler results in a successful upgrade, but any query to evmutil params (current or historical) fails. presumably if the params were actually used the chain itself may panic as well (or at least no txs/queries using them would work) 😄

i am working on the setup of migrations in evmutil and add in a param with consensus version bump to this pr.

**Review by galxy25 — APPROVED** [review]

**Review by drklee3 — APPROVED** [review]

**Pull request merged**